src/staker.cairo

```diff
@@ -96,6 +107,13 @@ pub mod Staker {
         }
     }
 
+    #[derive(Drop, Serde, starknet::Store)]
+    struct StakingLogRecord {
+        timestamp: u64,
+        total_staked: u128,
```

Comment:
probably better to store `total_staked * seconds elapsed` accumulator so we can get  time weighted average total staked over a period. it makes the value ~32 bits wider but you can still always get the exact total staked amount at a particular time by using the time weighted average over 1 second

Reply:
the reason is that the time weighted total staked is what counts as the total voting weight, not the total staked at a particular second